stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/EventsParser.kt

```diff
@@ -100,7 +110,13 @@ internal class EventsParser(
                     onSocketError(ChatNetworkError.create(ChatErrorCode.CANT_PARSE_CONNECTION_EVENT))
                 }
             } else {
-                onEvent(event)
+                // Don't call onEvent() immediately, instead
+                // let the caching class decide when it needs to be called
+                if (event is TypingStartEvent || event is TypingStopEvent) {
+                    typingEventCache.processEvent(event)
+                } else {
+                    onEvent(event)
+                }
```

Comment:
What if customers rely on `TypingStartEvent` to keep coming.
Shouldn't we just fire `TypingStopEvent` on timeout and not prevent `TypingStartEvent` from being passed further to avoid any breaking changes?

Reply:
That is essentially what is done [here](https://github.com/GetStream/stream-chat-android/pull/3637/files/cf3e25d9cceb470877d3c90539d6b60214bd6c82#diff-d1cedbe1ab0f946234b94756db4e441dc2fc5b7f712964c1941f9cc839f6f184R77).

You'll notice that the event is processed by basically getting registered and stored and then straight fired back out.

The reason why we are processing it is not to hijack it but rather to register it so we can time it's `TypingStopEvent` counterpart.